utils/dataHandling.ts

```diff
@@ -0,0 +1,277 @@
+    
+    console.log('[PROFILE] Final read relays:', readRelays);
+
+    const store = deriveProfile(pubkey);
+    console.log('[PROFILE] Store created:', store);
+    
+    const timeoutMs = 10000;
+    const pollInterval = 200;
+    let done = false;
+
+    const profile = await new Promise<any>((resolve, reject) => {
+      const start = Date.now();
+      let unsub: (() => void) | undefined; 
+      let timer: number | undefined; 
+
+      unsub = store.subscribe((value) => {
+        console.log('[PROFILE] Store callback received:', value);
+        console.log('[PROFILE] Value type:', typeof value);
+        console.log('[PROFILE] Value keys:', value ? Object.keys(value) : 'null/undefined');
+        console.log('[PROFILE] Full value:', JSON.stringify(value, null, 2));
+        
+        if (value && (value.name || value.display_name || value.about)) {
+          if (!done) {
+            done = true;
+            console.log('[PROFILE] Resolved profile:', value);
+            unsub?.(); 
+            clearInterval(timer);
+            resolve(value);
+          }
+        }
+      });
+
+      timer = setInterval(() => {
+        if (Date.now() - start > timeoutMs) {
+          if (!done) {
+            done = true;
+            console.warn('[PROFILE] Timeout waiting for profile');
+            unsub?.();
+            clearInterval(timer);
+            reject(new Error('Timed out waiting for profile'));
+          }
+        }
+      }, pollInterval);
+    });
+
+    return profile;
+  } catch (error: any) {
+    console.error('[PROFILE] Error loading profile:', error);
+    throw new Error(error.message || 'Failed to load profile');
+  }
+}; 
+
+/**
+ * Updates the user's profile name by publishing a kind 0 metadata event
+ * @param newName - The new name to set
+ * @param currentProfile - Optional current profile data to avoid reloading
+ * @returns Promise that resolves when the event is published
+ */
+export const changeName = async (newName: string, currentProfile?: any): Promise<void> => {
+  if (!newName || newName.trim() === '') {
+    throw new Error('Name cannot be empty');
+  }
+  try {
+    console.log('[PROFILE] Updating name to:', newName);
+    
+    const currentPubkey = pubkey.get();
+    if (!currentPubkey) {
+      throw new Error('No user logged in');
+    }
+
+    //get total profile data so updating doesnt lose anything
+    let profileData = currentProfile;
+    if (!profileData) {
+      profileData = await loadUserProfile(currentPubkey);
+    }
+    console.log('[PROFILE] Current profile data:', profileData);
+
+    const profileContent = JSON.stringify({
+      ...profileData, // Preserve existing fields
+      name: newName.trim(), // Update only the name
+    });
+    
+    const profileEvent = makeEvent(0, { content: profileContent });
+
+    // Get relays to publish to
+    const router = Router.get();
+    const publishRelays = router.FromUser().getUrls();
+    
+    if (publishRelays.length === 0) {
```

Comment:
This will never happen, since the router automatically falls back to the default relays.